caddyhttp/header/header_test.go

```diff
@@ -50,6 +52,8 @@ func TestHeader(t *testing.T) {
 		}
 
 		rec := httptest.NewRecorder()
+		// preset header
+		rec.Header().Set("Server", "Caddy")
```

Comment:
Just want to confirm that the headers set before `header.HTTP` are removed after all.